### PR TITLE
Vehicle similarity search module (ETA components)

### PR DESCRIPTION
**Pull request by chrisstauffer**
common core components to first pass encoding and matching capabilities.

### CODE REVIEW

**Comment by jasoncorso** [comment]
Is this now to be reviewed?

File: eta/core/features.py [hunk 1]
```diff
@@ -25,118 +25,168 @@
...
+
+        Args:
+            data: the data to featurize
+
+        Returns:
+            the feature vector
+        '''
         self.start(warn_on_restart=False, keep_alive=False)
-        v = self._featurize(data)
+        fv = self._featurize(data)
```
Comment (jasoncorso) [inline eta/core/features.py]: 2-char variable names are just as bad as 1-char variable names!  And I am fine with either in cases like this.

File: eta/core/numutils.py [hunk 2]
```diff
@@ -71,12 +71,21 @@ class GrowableArray(object):
...
+        Args:
+            rowlen: the desired length of each row
+        '''
         self.data = []
         self.rowlen = rowlen
 
     def update(self, row):
         '''Add row to array.'''
-        assert len(row) == self.rowlen, "Expected row length %d" % self.rowlen
+        if len(row) != self.rowlen:
```
Comment (jasoncorso) [inline eta/core/numutils.py]: Unclear to me what the difference is in functionality or error-checking.

File: eta/core/types.py [hunk 3]
```diff
@@ -676,3 +718,195 @@ class FrameSequence(JSONFileSequence):
...
+    '''A sequence of features indexed by one numeric parameter.
+
+    Examples:
+        /path/to/features/%05d.npz
+    '''
+
+    @staticmethod
+    def gen_path(basedir, params):
+        return os.path.join(
+            basedir, "{name}", "{idx}.npz").format(**params)
```
Comment (jasoncorso) [inline eta/core/types.py]: issue should be created to allow generalization beyond just npz here and elsewhere.

File: eta/core/numutils.py (hunk 2 above)
Comment (brimoor) [reply eta/core/numutils.py]: just following the developing convention to raise custom exceptions that categorize our errors

File: eta/core/types.py (hunk 3 above)
Comment (brimoor) [reply eta/core/types.py]: Yeah and here we'd create sub-types of `Features` to encapsulate the various implementations

**Pull request merged**